_includes/manuals/2.4/3.6_upgrade.md

```diff
@@ -31,113 +52,138 @@ For more information about how to backup your instance head over to
-get the correct one.
+Foreman runs best on PostgreSQL 12 but the default version on EL8 is 10. It is recommended to upgrade by [switching module streams](https://docs.fedoraproject.org/en-US/modularity/using-modules-switching-streams/).
 
-1. Go to the Foreman root directory and run the following command:
+First check which version of PostgreSQL is currently installed:
 
-For staying on the stable branch:
+{% highlight bash %}
+rpm -qv postgresql-server
+{% endhighlight %}
 
-- git checkout {{page.version}}-stable
-- git pull
+If you're already on 12 or don't have the PostgreSQL server installed, you can continue with step 3. Otherwise upgrade:
 
-There may have been changes to the ruby gem or node module versions, to update
-run the following commands:
+{% highlight bash %}
+systemctl stop postgresql.service
+# https://bugzilla.redhat.com/1935301
+sed -i '/^data_directory =/d' /var/lib/pgsql/data/postgresql.conf
+dnf module reset postgresql
+dnf module install postgresql:12
+dnf install postgresql-upgrade
+postgresql-setup --upgrade
+systemctl start postgresql.service
+{% endhighlight %}
+</div>
 
-    bundle update
-    npm update
+<div class="upgrade_os upgrade_os_debian10 upgrade_os_ubuntu1804">
+Upgrading from the last release to {{page.version}} has been tested. Updating
+the packages will upgrade the application and automatically migrate the
+database.
 
-The following step is the one that could change the contents of your database.
-Go to your new Foreman directory (or the git dir), then migrate and update the
-contents of your database:
-
-    foreman-rake db:migrate
-    foreman-rake db:seed
+First edit `/etc/apt/sources.list.d/foreman.list` and change the release
+number from the previous release to `{{ page.version }}`:
+
+<div class="upgrade_os upgrade_os_debian10">
+{% highlight bash %}
+deb http://deb.theforeman.org/ buster {{ page.version }}
+deb http://deb.theforeman.org/ plugins {{ page.version }}
+{% endhighlight %}
+</div>
+<div class="upgrade_os upgrade_os_ubuntu1804">
+{% highlight bash %}
+deb http://deb.theforeman.org/ bionic {{ page.version }}
+deb http://deb.theforeman.org/ plugins {{ page.version }}
+{% endhighlight %}
+</div>
 
-You should compile i18n strings and precompile assets now:
+Next upgrade all Foreman packages:
 
-    foreman-rake locale:pack
-    foreman-rake assets:precompile RAILS_ENV=production
-    foreman-rake webpack:compile RAILS_ENV=production
+{% highlight bash %}
+apt-get update
+apt-get --only-upgrade install ruby\* foreman\*
+{% endhighlight %}
+</div>
 
 #### Step 3 - Post-upgrade steps
 
 ##### Step 3 (A) - Database migration and cleanup
 
-The database should be migrated already, but you can make sure by executing the
-migration script again, it should produce no errors or output:
+Make sure by executing database is migrated. It should produce no errors or
+output:
 
     foreman-rake db:migrate
     foreman-rake db:seed
```

Comment:
Don't we run migrations from the installer and seeds on startup? this should be optional probably